App.js

```diff
@@ -1,9 +1,363 @@
+    { key: 2, name: '', invisible: true },
+    { key: 3, name: '', invisible: true },
+    { key: 4, name: '', invisible: true },
+    { key: 5, name: '', invisible: true },
+    { key: 6, name: 'Score' },
+    { key: 7, name: 0 },
+  ];
+
+  const numColumns = 7;
+
+  // render Nav Bar
+  const renderNav = ({ item }) => {
+    if (item.invisible === true) {
+      return <View style={[styles.item, styles.itemInvisible]} />;
+    }
+    if (item.name === 'PLAY') {
+      return (
+        <View style={styles.item}>
+          <TouchableHighlight onPress={() => pressPlay()}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    if (item.key === 7) {
+      return (
+        <View style={styles.item}>
+          <Text style={styles.itemText}>{score}</Text>
+        </View>
+      );
+    }
+    return (
+      <View style={styles.item}>
+        <Text style={styles.itemText}>{item.name}</Text>
+      </View>
+    );
+  };
+
+  // render board
+  const renderBoard = ({ item }) => {
+    if (item.invisible === true) {
+      return <View style={[styles.item, styles.itemInvisible]} />;
+    }
+    // Show ROLL button
+    if (item.name === 'ROLL') {
+      return (
+        <View style={styles.item}>
+          <TouchableHighlight onPress={() => pressRoll(item.key)}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    // Create Even button and mark it if it has been selected
+    if (item.name === 'EVEN') {
+      return (
+        <View style={even ? [styles.item, styles.markSpot] : styles.item}>
+          <TouchableHighlight onPress={() => pressEven()}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    // Create Odd button and mark it if it has been selected
+    if (item.name === 'ODD') {
+      return (
+        <View style={odd ? [styles.item, styles.markSpot] : styles.item}>
+          <TouchableHighlight onPress={() => pressOdd()}>
+            <Text style={styles.itemText}>{item.name}</Text>
+          </TouchableHighlight>
+        </View>
+      );
+    }
+    // Show ROLL NUMBER
+    if (item.key === 11) {
+      return (
+        <View style={styles.item}>
+          <Text style={styles.itemText}>{roll}</Text>
+        </View>
+      );
+    }
+    // mark current spot on board
+    if (position === item.boardNumber && (item.invisible === undefined || item.invisible === false)) {
+      return (
+        <View style={[styles.item, styles.markSpot]}>
+          <Text style={styles.itemText}></Text>
+        </View>
+      );
+    }
+    return (
+      <View style={styles.item}>
+        <Text style={styles.itemText}>{item.name}</Text>
+      </View>
+    );
+  };
+
+  // press Play button
+  const pressPlay = useCallback(() => {
+    setGameState(() => {
+      return {
+        roll: null,
```

Comment:
Should roll start as `null` or `0`? It's `null` here but `0` in your default state?

Reply:
Fixed to be 0.